6J1tQbLS Creating Roles and set permissions

41hulk opened the pull request.

mansurissa committed:
ch-setup(initial setup): setting up project

- install express
- install dotenv
- making test route

GuevaraMarius committed:
feat(authentication):signin

- create user model and migration
- create seeder users
- hash password
- sign in
- asign token to signed in user

chore:(eslint): configuringneslint

- install and configure eslint
- using airbnb base style

chore: connect-to-db
- creating seeds
- creating migration
- user model

 bg-fixing some model error

ft: signin

- registering user
- validations with join

 hashing pass2word before saving in db

feat(authenticatiom):sign in

- login for user
- signing token to signed in user

KundwaBruno committed:
Using google cloud transalte api for supporting multi languages

Changed to ES6 syntax

KundwaBruno committed:
Fix issue #15

Fixed Issue_#15'

issue_#15

Fixed issue #15 and i18n setup

xUjJi1sq Chore Support multiple languages

41hulk committed:
ch-CRUD-role

CREATE Roles
READ Roles
UPDATE Roles
DELETE Roles

41hulk committed:
ch-permissions-middleware

ghost commented:
## DeepCode's analysis on [#c85934](https://www.deepcode.ai/app/gh/atlp-rwanda/incredible-phantom-backend/ded2ba2399cc7a31a4458bf52b1abb247b86ec4f/atlp-rwanda/incredible-phantom-backend/c859344bf1eca7c89e7d3c73e0832903476c7684/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) found:

- :information_source: **1** minor issue. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Missing JSDoc for parameter 'models'. Occurrences: <ul><li><a href="https://github.com/atlp-rwanda/incredible-phantom-backend/blob/c859344bf1eca7c89e7d3c73e0832903476c7684/src/database/models/user.js#L4">user.js:4</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/atlp-rwanda/incredible-phantom-backend/ded2ba2399cc7a31a4458bf52b1abb247b86ec4f/atlp-rwanda/incredible-phantom-backend/c859344bf1eca7c89e7d3c73e0832903476c7684/pr/_/%2Fsrc%2Fdatabase%2Fmodels%2Fuser.js/javascript%2Feslint%2Fvalid-jsdoc/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/atlp-rwanda/incredible-phantom-backend/ded2ba2399cc7a31a4458bf52b1abb247b86ec4f/atlp-rwanda/incredible-phantom-backend/c859344bf1eca7c89e7d3c73e0832903476c7684/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=atlp-rwanda)

41hulk committed:
Fix-issue-package-lock

41hulk committed:
feat(Roles & Permissions): Setting Roles and Permissions

-Admin can create,update,delete roles
-Admin can set and update Roles's permission

[Delivers #P4Z442wG]
- Ntare Guy -
